.github/workflows/build.yml

```diff
@@ -8,258 +11,113 @@ on:
-    steps:
-      # - uses: gardenlinux/workflow-telemetry-action@c75b594f552d305ffd5f9074637137bc343ba35e # pin@v2
-      #   with:
-      #     metric_frequency: 1
-      #     proc_trace_min_duration: 10
-      #     proc_trace_chart_max_count: 50
-      #     comment_on_pr: false
-      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # pin@v4.1.1
-        with:
-          submodules: true
-      - uses: ./.github/actions/setup
-      - name: set VERSION=${{ needs.version.outputs.version }}
-        run: |
-          cd $(dirname $(realpath bin/garden-version))/..
-          bin/garden-version "${{ needs.version.outputs.version }}" | tee VERSION
-          git update-index --assume-unchanged VERSION
-      - name: build certificates
-        if: ${{ ! inputs.use_kms }}
-        run: ./cert/build
-      - name: use kms backed certificates
-        if: ${{ inputs.use_kms }}
-        run: |
-          touch cert/gardenlinux-secureboot.db.arn
-          for f in secureboot.{{pk,null.pk,kek,db}.{crt,der,auth},db.arn,aws-efivars}; do
-            ln -sr "cert/gardenlinux-$f" "cert/$f"
-          done
-      - uses: actions/cache/save@d4323d4df104b026a6aa633fdb11d772146be0bf # pin@v4.2.2
-        with:
-          path: cert/*.*
-          key: cert-${{ github.run_id }}
-  base:
-    name: bootstrap stage
-    needs: version
-    runs-on: ${{ matrix.arch == 'arm64' && 'ubuntu-latest-arm' || 'ubuntu-latest' }}
-    defaults:
-      run:
-        shell: bash
-    strategy:
-      fail-fast: false
-      matrix:
-        arch: [ amd64, arm64 ]
-    steps:
-      # - uses: gardenlinux/workflow-telemetry-action@c75b594f552d305ffd5f9074637137bc343ba35e # pin@v2
-      #   with:
-      #     metric_frequency: 1
-      #     proc_trace_min_duration: 10
-      #     proc_trace_chart_max_count: 50
-      #     comment_on_pr: false
-      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # pin@v4.1.1
-        with:
-          submodules: true
-      - uses: ./.github/actions/setup
-        with:
-          arch: "${{ matrix.arch }}"
-      - name: set VERSION=${{ needs.version.outputs.version }}
-        run: |
-          cd $(dirname $(realpath bin/garden-version))/..
-          bin/garden-version "${{ needs.version.outputs.version }}" | tee VERSION
-          git update-index --assume-unchanged VERSION
-      - name: build base-${{ matrix.arch }}
-        run: make base-${{ matrix.arch }}-build
-      - uses: actions/cache/save@d4323d4df104b026a6aa633fdb11d772146be0bf # pin@v4.2.2
-        with:
-          path: .build
-          key: base-${{ matrix.arch }}-${{ github.run_id }}
-  generate_matrix_images:
-    uses: ./.github/workflows/generate_matrix.yml
+  requirements:
+    name: Prepare build requirements
+    uses: ./.github/workflows/build_requirements.yml
+    with:
+      version: ${{ inputs.version || 'now' }}
+      use_kms: ${{ github.event_name == 'workflow_dispatch' && inputs.use_kms || false }}
```

Comment:
with this, use_kms would only be true if this workflow is manually triggered (workflow_dispatch). Do we want this? 


Previous logic was defined in nightly.yml 
```
use_kms: ${{ github.event_name != 'workflow_dispatch' || inputs.use_kms }}
```

Reply:
Good catch but from a different perspective: `build.yml` is a reusable workflow and not callable at all. The logic above would always result in `false` therefore.